docs: translate `renderToString` section (#496)
reactjs/id.react.dev: src/content/reference/react-dom/server/renderToString.md

<Pitfall>

> `renderToString` tidak mendukung streaming atau menunggu data. [Lihat alternatif.](#alternatives)

Reviewer: ```suggestion
`renderToString` tidak mendukung *streaming* atau menunggu data. [Lihat beberapa alternatifnya.](#alternatives)
```

Reviewer: Please don't forget to italicize the `streaming` word, @hisamafahri. Thanks!

Author: Ah, missed that one. Should be fixed now

---

## Referensi {/*reference*/}

### `renderToString(reactNode)` {/*rendertostring*/}

Di server, panggil `renderToString` untuk me-*render* aplikasi Anda ke HTML.

```js
import { renderToString } from 'react-dom/server';

const html = renderToString(<App />);
```

Di klien, panggil [`hydrateRoot`](/reference/react-dom/client/hydrateRoot) untuk membuat HTML yang dibuat server interaktif.

[Lihat lebih banyak contoh di bawah.](#usage)

#### Parameter {/*parameters*/}

* `reactNode`: Node React yang ingin Anda render ke HTML. Contohnya, sebuah elemen JSX seperti `<App />`

#### Kembalian {/*returns*/}

Sebuah *string* HTML.

#### Peringatan {/*caveats*/}

* `renderToString` memiliki dukungan *Suspense* yang terbatas. Jika komponen ditangguhkan, `renderToString` secara langsung mengirimkan kembaliannya sebagai HTML.

> * `renderToString` bejerka di peramban, tapi menggunakannya di kode klien [tidak dirokemndasikan.](#removing-rendertostring-from-the-client-code)

Reviewer 2: Typo
```suggestion
* `renderToString` bekerja di peramban, tapi menggunakannya di kode klien [tidak direkomendasikan.](#removing-rendertostring-from-the-client-code)
```

---

## Penggunaan {/*usage*/}

### Me-render pohon React sebagai HTML menjadi string {/*rendering-a-react-tree-as-html-to-a-string*/}

> Panggil `renderToString` untuk me-render aplikasi Anda ke *string* HTML yang dapat Anda kirim dengan respons server Anda:

Reviewer 2: ```suggestion
Panggil `renderToString` untuk me-*render* aplikasi Anda ke *string* HTML yang dapat Anda kirim dengan respons server Anda:
```

```js {5-6}
import { renderToString } from 'react-dom/server';

> // Sintaks pengendali rute bergantung pada framework backend Anda

Reviewer 2: ```suggestion
// Sintaksis pengendali rute bergantung pada framework backend Anda
```

app.use('/', (request, response) => {
  const html = renderToString(<App />);
  response.send(html);
});
```

Ini akan menghasilkan output HTML non-interaktif awal dari komponen React Anda. Pada klien, Anda perlu memanggil [`hydrateRoot`](/reference/react-dom/client/hydrateRoot) untuk *menghidrasi* HTML yang dihasilkan server dan membuatnya interaktif.

<Pitfall>

`renderToString` tidak mendukung *streaming* atau menunggu data. [Lihat beberapa alternatifnya.](#alternatives)

</Pitfall>

---

## Alternatif {/*alternatives*/}

### Migrasi dari `renderToString` ke metode streaming di server {/*migrating-from-rendertostring-to-a-streaming-method-on-the-server*/}

> `renderToString` segera mengembalikan *string*, sehingga tidak mendukung *streaming* atau menunggu data.

Reviewer 2: Not sure about this.

```suggestion
`renderToString` mengembalikan *string* dengan seketika, sehingga tidak mendukung *streaming* atau menunggu data.
```

Author: Looks good, I'll take it 👍

> Jika memungkinkan, sebaiknya gunakan alternatif berfitur lengkap ini:

Reviewer 2: ```suggestion
Saat memungkinkan, kami merekomendasikan untuk menggunakan alternatif yang berfitur lengkap ini:
```

* Jika Anda menggunakan Node.js, gunakan [`renderToPipeableStream`.](/reference/react-dom/server/renderToPipeableStream)
* Jika Anda menggunakan Deno atau _edge runtime_ modern dengan [Web Streams](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API), gunakan [`renderToReadableStream`.](/reference/react-dom/server/renderToReadableStream)

Anda dapat terus menggunakan `renderToString` jika _environment_ server Anda tidak mendukung _streaming_.

---

### Menghapus `renderToString` dari kode klien {/*removing-rendertostring-from-the-client-code*/}

Terkadang, `renderToString` digunakan pada klien untuk mengkonversi beberapa komponen ke HTML.

```js {1-2}
// 🚩 Tidak perlu: menggunakan renderToString pada klien
import { renderToString } from 'react-dom/server';

const html = renderToString(<MyIcon />);
console.log(html); // Contohnya, "<svg>...</svg>"
```

> Mengimpor `react-dom/server` **pada klien** meningkatkan ukuran bundel Anda secara tidak perlu dan harus dihindari. Jika Anda perlu me-_render_ beberapa komponen ke HTML di browser, gunakan [`createRoot`](/reference/react-dom/client/createRoot) dan baca HTML dari DOM:

Reviewer 3: ```suggestion
Mengimpor `react-dom/server` **pada klien** meningkatkan ukuran bundel Anda secara tidak perlu dan harus dihindari. Jika Anda perlu me-*render* beberapa komponen ke HTML di peramban, gunakan [`createRoot`](/reference/react-dom/client/createRoot) dan baca HTML dari DOM:
```

flushSync(() => {
  root.render(<MyIcon />);
});
console.log(div.innerHTML); // Contohnya, "<svg>...</svg>"
```

> Panggilan [`flushSync`](/reference/react-dom/flushSync) diperlukan agar DOM diperbarui sebelum membaca [`innerHTML`](https://developer.mozilla.org/en-US/docs/Web/API/Elemen/innerHTML).

Reviewer 2: ```suggestion
Memanggil [`flushSync`](/reference/react-dom/flushSync) diperlukan agar DOM diperbarui sebelum membaca [`innerHTML`](https://developer.mozilla.org/en-US/docs/Web/API/Elemen/innerHTML).
```

---

## Penyelesaian Masalah {/*troubleshooting*/}

### Saat komponen ditangguhkan, HTML selalu berisi fallback {/*when-a-component-suspends-the-html-always-contains-a-fallback*/}

`renderToString` tidak sepenuhnya mendukung *Suspense*.

> Jika beberapa komponen ditangguhkan (misalnya, karena ditentukan dengan [`lazy`](/reference/react/lazy) atau mengambil data), `renderToString` tidak akan menunggu kontennya diselesaikan. Sebagai gantinya, `renderToString` akan menemukan batas [`<Suspense>`](/reference/react/Suspense) terdekat di atasnya dan merender prop `fallback` di HTML. Konten tidak akan muncul hingga kode klien dimuat.

Reviewer: ```suggestion
Jika beberapa komponen ditangguhkan (misalnya, karena ditentukan dengan [`lazy`](/reference/react/lazy) atau mengambil data), `renderToString` tidak akan menunggu kontennya diselesaikan. Sebagai gantinya, `renderToString` akan menemukan batas [`<Suspense>`](/reference/react/Suspense) terdekat di atasnya dan merender prop `fallback` di HTML. Konten tidak akan muncul hingga kode klien dimuat.

```

Please restore the empty lines. The empty lines are necessary to split paragraphs in Markdown.

Author: I am not really sure why GitHub display it as a deleted line. But I don't delete it 😄 👍 
<img width="1350" alt="image" src="https://user-images.githubusercontent.com/65691613/235587938-ade3e0ba-a784-46d6-bafd-aa849a8db1e3.png">

Reviewer: Oops, my bad. Thanks for clarifying. 😅
